doc/rest-api.md

```diff
@@ -1818,3 +1818,287 @@ Input (none at present):
+ * Operation: sync
+ * Return: dict representing a storage pool
+
+    {
+        "type": "sync",
+        "status": "success",
+        "status_code": 200,
+        "operation": "",
+        "error_code": 0,
+        "error": "",
+        "metadata": {
+            "pool_name": "pool1",
+            "driver": "lvm",
+            "used_by": [
+                "/1.0/containers/alp1",
+                "/1.0/snapshots/alp1/snap0",
+                "/1.0/snapshots/alp1/snap1",
+                "/1.0/snapshots/alp1/snap2",
+                "/1.0/containers/alp2",
+                "/1.0/containers/alp3",
+                "/1.0/images/cedce20b5b236f1071134beba7a5fd2aa923fda49eea4c66454dd559a5d6e906",
+                "/1.0/custom/vol1"
+                ],
+            "config": {
+                "size": "15032385536",
+                "source": "pool1",
+                "volume.block.filesystem": "xfs",
+                "volume.block.mount_options": "discard",
+                "volume.lvm.thinpool_name": "lxdpool",
+                "volume.size": "10737418240"
+            }
+        }
+    }
+
+
+### POST
+
+ * Description: rename a storage pool
+ - Currently not implemented.
+
+### PUT (ETag supported)
+
+ * Description: replace the storage pool information
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+ Input:
+
+    {
+        "config": {
+            "size": "15032385536",
+            "source": "pool1",
+            "volume.block.filesystem": "xfs",
+            "volume.block.mount_options": "discard",
+            "volume.lvm.thinpool_name": "LXDPool",
+            "volume.size": "10737418240"
+        }
+    }
+
+### PATCH
+
+ * Description: update the storage pool configuration
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {
+            "volume.block.filesystem": "xfs",
+        }
+    }
+
+### DELETE
+
+ * Description: delete a storage pool
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input (none at present):
+
+    {
+    }
+
+## /1.0/storage-pools/<pool_name>/volumes
+
+### GET
+
+ * Description: list of storage volumes
+ * Introduced: with API extension "storage"
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of storage volumes that currently exist on a given storage pool
+
+    [
+        {
+            "volume_type": "container",
+            "used_by": [],
+            "volume_name": "alp1/snap1",
+            "config": {}
+        },
+        {
+            "volume_type": "container",
+            "used_by": [],
+            "volume_name": "alp1/snap2",
+            "config": {}
+        },
+    ]
+
+## /1.0/storage-pools/<pool_name>/volumes/<volume_type>
```

Comment:
shouldn't that still be under "/1.0/storage-pools/<pool_name>/volumes" without "/<volume_type>"?

We should have a github issue to add GET/POST to "/1.0/storage-pools/<pool_name>/volumes/<volume_type>" later though as a way to just list/create a particular volume type.